codegen/datamodelgen.py

```diff
@@ -105,16 +107,19 @@ class DataModelGenerator:
     def __init__(self):
         self.version = get_version_for_filepath()
         self._static_info: Dict[str, DataModelStaticInfo] = {
-            "workflow": DataModelStaticInfo("workflow", "meshing", self.version),
-            "meshing": DataModelStaticInfo("meshing", "meshing", self.version),
+            "workflow": DataModelStaticInfo("workflow", ("meshing",), self.version),
+            "meshing": DataModelStaticInfo("meshing", ("meshing",), self.version),
             "PartManagement": DataModelStaticInfo(
-                "PartManagement", "meshing", self.version
+                "PartManagement", ("meshing",), self.version
             ),
             "PMFileManagement": DataModelStaticInfo(
-                "PMFileManagement", "meshing", self.version
+                "PMFileManagement", ("meshing",), self.version
             ),
             "icing": DataModelStaticInfo(
-                "flserver", "flicing", self.version, "flicing"
+                "flserver", ("flicing",), self.version, "flicing"
+            ),
+            "preferences": DataModelStaticInfo(
+                "preferences", ("meshing", "solver", "flicing,"), self.version
             ),
```

Comment:
@hpohekar here we will add an entry for solver_workflow with modes = ("solver",)